gbpservice/neutron/tests/unit/nfp/configurator/lib/test_schema_validator.py

```diff
@@ -0,0 +1,204 @@
+        self.assertTrue(result)
+
+    def test_validate_schema_for_request_data_config(self):
+        """Test case to test validate_schema() of schema_validator.py for
+           'request_data_config' schema
+        """
+        result = self.sv.validate_schema(self.sr.request_data_config,
+                                         schema.request_data_config)
+        self.assertTrue(result)
+
+    def test_validate_schema_for_interfaces(self):
+        """Test case to test validate_schema() of schema_validator.py for
+           'interfaces' schema
+        """
+        result = self.sv.validate_schema(self.sr.interfaces,
+                                         schema.interfaces)
+        self.assertTrue(result)
+
+    def test_validate_schema_for_interfaces_rule_info(self):
+        """Test case to test validate_schema() of schema_validator.py for
+           'interfaces_rule_info' schema
+        """
+        result = self.sv.validate_schema(self.sr.interfaces_rule_info,
+                                         schema.interfaces_rule_info)
+        self.assertTrue(result)
+
+    def test_validate_schema_for_routes(self):
+        """Test case to test validate_schema() of schema_validator.py for
+           'routes' schema
+        """
+        result = self.sv.validate_schema(self.sr.routes,
+                                         schema.routes)
+        self.assertTrue(result)
+
+    def test_validate_schema_for_healthmonitor(self):
+        """Test case to test validate_schema() of schema_validator.py for
+           'healthmonitor' schema
+        """
+        result = self.sv.validate_schema(self.sr.healthmonitor,
+                                         schema.healthmonitor)
+        self.assertTrue(result)
+
+    def test_decode_for_interfaces(self):
+        """Test case to test decode() of schema_validator.py for request_data
+           with resource 'interfaces'
+        """
+        request_data = self.make_request_data(self.sr.resource_interfaces,
+                                              self.sr.interfaces)
+        result = self.sv.decode(request_data)
+        self.assertTrue(result)
+
+    def test_decode_for_routes(self):
+        """Test case to test decode() of schema_validator.py for request_data
+           with resource 'routes'
+        """
+        request_data = self.make_request_data(self.sr.resource_routes,
+                                              self.sr.routes)
+        result = self.sv.decode(request_data)
+        self.assertTrue(result)
+
+    def test_decode_for_healthmonitor(self):
+        """Test case to test decode() of schema_validator.py for request_data
+           with resource 'healthmonitor'
+        """
+        request_data = self.make_request_data(self.sr.resource_healthmonitor,
+                                              self.sr.healthmonitor)
+        result = self.sv.decode(request_data)
+        self.assertTrue(result)
+
```

Comment:
Yes, decode is invoked for *aaS api also. Will add one more test case to cover this